xtevent/_eventgenvars.ado

```diff
@@ -1,306 +1,500 @@
+	
+	impute(string) /*impute policyvar. There are three options: */
+	/*nuchange  imputes outer missing values of policyvar without verifying staggered adoption*/
+	/*stag  imputes outer missing values of policyvar verifying staggered adoption*/
+	/*instag  imputes outer and inner missing values verifying staggered adoption*/
+	STatic /* Estimate static model */
+	
+
+	
+	]	
+	;
+	#d cr	
+	
+	tempvar mz mz2 kg
+	
+	marksample touse
+	
+	* mz maximum of policy outside window
+	* kg grouped event time variable, grouping dummies outside window
+	
+	loc z = "`policyvar'"
+
+	if "`static'"==""{
+		* Get span of calendar time
+		qui su `timevar' if `touse', d
+		loc tmin=r(min)
+		loc tmax=r(max)
+		loc tdiff = `tmax'-`tmin'
+		
+		* Using notation from latest draft: https://jorgeperezperez.com/files/EventStudy.pdf 
+		
+		* Endpoints are left: G+L+1
+		* Right: M+1
+		
+		* Left window is G+L
+		* Right window is M 
+		
+		* If you have t=-3 -2 -1 0 1 2 3, tdiff=6 then the maximum window size you can have is 2 on each side
+		* If you have t=-3 -2 -1 0 1 2 3 4, tdiff=7 then the maximum window size you can have is 2 left, 3 right
+			
+		
+		* Error check for window outside event time range
+		if abs(`lwindow') > `=abs(`tdiff')-1' | abs(`rwindow') > `=abs(`tdiff')-1' {
+			di as err _n "Window outside event-time range."
+			exit 301
+		}
+		
+		* Error check for trend event time range
+		if "`trend'"!="" {
+			if abs(`trend') > `=abs(`tdiff')-1' {
+				di as err _n "Trend outside time range."
+				exit 301
+			}
+					
+			* Error check for trend outside window
+			if abs(`trend')>abs(`lwindow') {
+				di as err _n "Trend outside window range."
+				exit 301
+			}
+		}
+		
+		* Check for vars named _k
+		cap unab oldkvars : _k*
+		if !_rc {
+			di as err _n "You have variable names with the _k prefix. _k is reserved for internal -xtevent- variables."
+			di as err _n "Please rename these variables before proceeding."
+			exit 110
+		}
+	}
+
+	********************* find first and last observed values *********************
+
+	*find minimum valid time (time where there is a no-missing observation)
+	tempvar zmint zmint2 zminv zminv2 zmaxt zmaxt2 zmaxv zmaxv2
+	cap by `panelvar' (`timevar'): egen `zmint'=min(`timevar') if !missing(`z') & `touse'	
+	cap by `panelvar' (`timevar'): egen `zmint2'=min(`zmint')
+	*find the corresponding minimum valid value
+	cap by `panelvar' (`timevar'): gen `zminv'=`z' if `timevar'==`zmint2' 
+	cap by `panelvar' (`timevar'): egen `zminv2'=min(`zminv')
+
+	*find maximum valid time
+	cap by `panelvar' (`timevar'): egen `zmaxt'=max(`timevar') if !missing(`z') & `touse'
+	cap by `panelvar' (`timevar'): egen `zmaxt2'=max(`zmaxt')
+	*find the corresponding maximum valid value
+	cap by `panelvar' (`timevar'): gen `zmaxv'=`z' if `timevar'==`zmaxt2' 
+	cap by `panelvar' (`timevar'): egen `zmaxv2'=max(`zmaxv')
+	
+	*create a copy of z. If imputation happens, it will be on this copy
+	cap drop zn2 
+	qui gen zn2=`z'
+	

```

Comment:
@Constantino-Carreto-Romero  Why do we need so many `captures` in this chunk of code? `capture` should only be used in cases where you expect some of these codes to return errors, and if that is the case you need to make sure that the errors are expected. If you are only trying to suppress output use `quietly`. `capture` is risky because we may miss unexpected errors.

Reply:
I compared which one was more appropriate, capture or quietly, and I changed it to quietly.